src/Utils.php

```diff
@@ -0,0 +1,380 @@
+<?php
+namespace GuzzleHttp;
+
+use GuzzleHttp\Exception\InvalidArgumentException;
+use GuzzleHttp\Handler\CurlHandler;
+use GuzzleHttp\Handler\CurlMultiHandler;
+use GuzzleHttp\Handler\Proxy;
+use GuzzleHttp\Handler\StreamHandler;
+use Psr\Http\Message\UriInterface;
+
+final class Utils
```

Comment:
Please me more defensive here. 

Mark it as `@internal` and  write a comment to explain the intent of the class.

Reply:
I don't think this class is internal only? We do intend for people to call these class methods, instead of the functions.

Reply:
Sure, It is just me that want to get rid of the functions AND eventually this class. 

We can discuss that in a separate PR or issue.